boards/imix/src/lowpan_frag_dummy.rs

```diff
@@ -399,41 +371,37 @@ impl<'a, A: time::Alarm + 'a> LowpanTest<'a, A> {
                                _: &[u8],
                                src_mac_addr: MacAddress,
                                dst_mac_addr: MacAddress) {
-        let frag_state = self.frag_state;
-        let tx_state = self.tx_state;
+        let frag_state = &self.frag_state;
         //frag_state.radio.config_set_pan(0xABCD);
         let ret_code = frag_state.transmit_packet(src_mac_addr,
```

Comment:
Is it not an issue that calls to transmit_packet() must be placed inside an unsafe block, because it requires that IP6_DGRAM have a static lifetime? I ask because I am currently trying to write a UDP capsule which will rely on this code (as will any code that implements Thread in a capsule), but I cannot use unsafe because I am writing capsule code. Or is there a way around this that I am not aware of?

Reply:
So I realized this is not an issue so long as I just have the IP6_DGRAM be passed in with a static lifetime from whatever kernel code ultimately calls my UDP capsule. Sorry for unnecessary comments, I am learning as I go.

Reply:
Exactly. In an _actual_ driver, you'll want to just have a field of `self` that has a `&'static` in the type already, and it'll get a static variable passed in (probably by the board config).